### DISCUSSION TITLE
[Question]: Gemini apparently not working via Vertex AI

### DISCUSSION THREAD

**Discussion by AsmardPing** [danny-avila/LibreChat#2819] [Troubleshooting]
### What is your question?

Hello,
I am trying to get the Google Endpoint running via the Vertex AI API. I followed the instructions from [this tutorial](https://www.librechat.ai/docs/configuration/pre_configured_ai/google). 
I have tried to provide the auth.json file both globally via the api/data folder (and I can see that it is correctly mounted within my docker container) and via the user_provided option. However, in both cases I receive the error:

> Something went wrong. Here's the specific error message we encountered: [GoogleClient] an API Key is required for Gemini models (Generative Language API)

I find the error to be confusing, as it indicates that Librechat is trying to connect to Google via the Generative Language API instead of using the Vertex AI API. Maybe I'm missing some setting to instruct it to use Vertex API?

Thank you in advance for any help!

### More Details

Librechat Version:
0.7.2

Google Settings within .env:
```
#============#
# Google     #
#============#

# GOOGLE_KEY=user_provided
# GOOGLE_REVERSE_PROXY=

# Gemini API
# GOOGLE_MODELS=gemini-1.0-pro,gemini-1.0-pro-001,gemini-1.0-pro-latest,gemini-1.0-pro-vision-latest,gemini-1.5-pro-latest,gemini-pro,gemini-pro-vision

# Vertex AI
GOOGLE_MODELS=gemini-1.5-pro-preview-0409,gemini-1.0-pro-vision-001,gemini-pro,gemini-pro-vision,chat-bison,chat-bison-32k,codechat-bison,codechat-bison-32k,text-bison,text-bison-32k,text-unicorn,code-gecko,code-bison,code-bison-32k

# Google Gemini Safety Settings
# NOTE (Vertex AI): You do not have access to the BLOCK_NONE setting by default.
# To use this restricted HarmBlockThreshold setting, you will need to either:
#
# (a) Get access through an allowlist via your Google account team
# (b) Switch your account type to monthly invoiced billing following this instruction:
#     https://cloud.google.com/billing/docs/how-to/invoiced-billing
#
# GOOGLE_SAFETY_SEXUALLY_EXPLICIT=BLOCK_ONLY_HIGH
# GOOGLE_SAFETY_HATE_SPEECH=BLOCK_ONLY_HIGH
# GOOGLE_SAFETY_HARASSMENT=BLOCK_ONLY_HIGH
# GOOGLE_SAFETY_DANGEROUS_CONTENT=BLOCK_ONLY_HIGH
```

docker-compose.override.yml:
```
services:
  # USE GOOGLE GEMINI FOR ALL USERS
  api:
    volumes:
      - type: bind
        source: ./api/data/auth.json
        target: /app/api/data/auth.json
```

### What is the main subject of your question?

Installation, Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
If you are providing the auth.json, remove the GOOGLE_KEY env var (or comment it out)

Same goes for the inverse (remove auth.json if doing the key)

Lastly, check if you have GOOGLE_API_KEY set, this is a deprecated value and will lead to conflicts. Remove and replace for GOOGLE_SEARCH_API_KEY

**Comment by AsmardPing** [comment]
Hi danny,
thanks for this extremely fast answer. The GOOGLE_KEY env var is commented out in my case (please refer to the "More Details" section above, it was not properly rendered when I created the question I updated it now). 

A value "GOOGLE_API_KEY" does not exist in my .env file, and the GOOGLE_SEARCH_API_KEY section looks like this:
```
# Google
#-----------------
GOOGLE_SEARCH_API_KEY=
GOOGLE_CSE_ID=
```
